app/assets/javascripts/ng-control/browser.js

```diff
@@ -323,47 +323,75 @@
       $scope.select_view('create');
     }
 
-    // Search
+    function sample_inventory(samples) {
+      return aq.collect(samples,function(s) {
+        var sample = new Sample($http).from(s);
+        if ( aq.url_params().sid && sample.id === parseInt(aq.url_params().sid) ) {
+          sample.open = true;
+          sample.inventory = true;
+          sample.loading_inventory = true;
+          sample.get_inventory(function() {
+            sample.loading_inventory = false;            
+            sample.inventory = true;
+          });              
+        }
+        return sample;
+      });
 
+    }
+
+    // Search
     $scope.search = function(p) {
```

Comment:
The `search` function should handle all of the cases that may occur, each of which depends on the state of the input fields.  Specifically, the item search should be called from here.